### ISSUE TITLE
Routing wrong action when using "controller:"

### ISSUE THREAD

**Issue by hexcat** [rails/rails#32493]
### Steps to reproduce

1. Create new application with `rails new`
2. Add `get 'wrong_action', to: :show, controller: 'test'` line to `config/routes.rb`
3. Create `app/controllers/test_controller.rb` and write this code into it:
```
class TestController < ApplicationController
  def show
    render plain: params
  end
end
```
4. Start server with `bin/rails server -b 127.0.0.1`
5. Navigate web browser to [http://127.0.0.1:3000/wrong_action](http://127.0.0.1:3000/wrong_action)

### Expected behavior
`{"controller"=>"test", "action"=>"show"}` [(Rails Routing from the Outside In)](http://guides.rubyonrails.org/routing.html#singular-resources)

### Actual behavior
`The action 'wrong_action' could not be found for TestController`

### System configuration
**Rails version**: 5.1.6

**Ruby version**: 2.5.1

**Comment by pixeltrix** [comment]
This behaviour was deprecated in cc26b6b7bccf0eea2e2c1a9ebdcc9d30ca7390d9 as part of Rails 4.2 and removed in 1f3b0a8609c00278b9a10076040ac9c90a9cc4a6 as part of Rail 5.0. Thanks for bringing this mistake in the guides to our attention. 👍

**Comment by pixeltrix** [comment]
Turns out that this was addressed by #28728 and will be fixed online once Rails 5.2 final is released.
